ui/js/controllers/perf/compare.js

```diff
@@ -116,31 +118,51 @@ perf.controller('CompareChooserCtrl', [
             };
 
             $scope.runCompare = function() {
-                ThResultSetModel.getResultSetsFromRevision($scope.originalProject.name, $scope.originalRevision).then(
-                    function() {
-                        $scope.originalRevisionError = undefined;
-                    },
-                    function(error) {
-                        $scope.originalRevisionError = error;
-                    }
-                );
-
-                ThResultSetModel.getResultSetsFromRevision($scope.newProject.name, $scope.newRevision).then(
-                    function () {
-                        $scope.newRevisionError = undefined;
-                        if ($scope.originalRevisionError === undefined && $scope.newRevisionError === undefined) {
-                            $state.go('compare', {
-                                originalProject: $scope.originalProject.name,
-                                originalRevision: $scope.originalRevision,
-                                newProject: $scope.newProject.name,
-                                newRevision: $scope.newRevision
-                            });
+                if($scope.originalRevision !== undefined){
```

Comment:
There's a lot of duplication here. Unfortunately the original code wasn't a great example to work from, so it's not your fault, but I think a better pattern would be to replace the existing code with a `$q.all` (https://docs.angularjs.org/api/ng/service/$q) and re-use the logic for getting new revision in both places. This would be a nice improvement over the old code as well.

E.g.:

```js
            $q.all([
                ThResultSetModel.getResultSetsFromRevision(
                    $scope.newProject.name, $scope.newRevision).then(function() {
                        $scope.newRevisionError = undefined;
                    },
                    function(error) {
                        $scope.newRevisionError = error;
                    }),
                ($scope.originalRevision) ? ThResultSetModel.getResultSetsFromRevision(
                    $scope.originalProject.name, $scope.originalRevision).then(function() {
                        $scope.originalRevisionError = undefined;
                    },
                    function(error) {
                        $scope.originalRevisionError = error;
                    })]).then(function() {
                        // we found result sets for both revisions, so have valid data
                        ...
                    });
```

Actually, looking at the history, it seems like I already asked you to do something like this in an earlier revision of the code. No matter, I know it's been a while since you've looked at this, but please implement this change before asking for another review.